src/AppInstallerCLICore/Workflows/MultiQueryFlow.cpp

```diff
@@ -0,0 +1,148 @@
+        {
+            auto& searchContext = *searchContextPtr;
+            SearchRequest searchRequest = searchContext.Get<Execution::Data::SearchRequest>();
+            searchContext.Add<Execution::Data::SearchResult>(searchContext.Get<Execution::Data::Source>().Search(searchRequest));
+
+            switch (m_searchPurpose)
+            {
+            case SearchPurpose::Install:
+                searchContext << Workflow::SelectSinglePackageVersionForInstallOrUpgrade(/* isUpgrade */ false);
+                break;
+            case SearchPurpose::Upgrade:
+                searchContext << Workflow::SelectSinglePackageVersionForInstallOrUpgrade(/* isUpgrade */ true);
+                break;
+            case SearchPurpose::Uninstall:
+                searchContext <<
+                    Workflow::HandleSearchResultFailures <<
+                    Workflow::EnsureOneMatchFromSearchResult(true) <<
+                    Workflow::ReportPackageIdentity;
+                break;
+            default:
+                THROW_HR(E_UNEXPECTED);
+            }
+
+            if (searchContext.IsTerminated())
+            {
+                if (context.IsTerminated() && context.GetTerminationHR() == E_ABORT)
+                {
+                    // This means that the subcontext being terminated is due to an overall abort
+                    context.Reporter.Info() << Resource::String::Cancelled << std::endl;
+                    return;
+                }
+                else
+                {
+                    auto packageString = GetPackageStringFromSearchRequest(searchRequest);
+                    auto searchTerminationHR = searchContext.GetTerminationHR();
+                    if (searchTerminationHR == APPINSTALLER_CLI_ERROR_UPDATE_NOT_APPLICABLE ||
+                        searchTerminationHR == APPINSTALLER_CLI_ERROR_PACKAGE_ALREADY_INSTALLED)
+                    {
+                        AICLI_LOG(CLI, Info, << "Package is already installed: [" << packageString << "]");
+                        context.Reporter.Info() << Resource::String::MultiInstallPackageAlreadyInstalled(packageString) << std::endl;
+                        continue;
+                    }
+                    else
+                    {
+                        AICLI_LOG(CLI, Info, << "Package not found: [" << packageString << "]");
+                        context.Reporter.Info() << Resource::String::MultiInstallSearchFailed(packageString) << std::endl;
+
+                        // Keep searching for the remaining packages and only fail at the end.
+                        foundAll = false;
+                        continue;
+                    }
+                }
+            }
+
+            packageSubContexts.emplace_back(std::move(searchContextPtr));
+        }
+
+        if (!foundAll)
+        {
+            AICLI_LOG(CLI, Info, << "Could not find one or more packages");
+            if (context.Args.Contains(Execution::Args::Type::IgnoreUnavailable))

```

Comment:
Does this arg need to be added to install / upgrade in case one of the multiqueries returns no packages? What happens if the user runs `winget install 7zip chrome ThisExampleIsNotPresent`? Would all packages fail, or would 7zip and chrome still be installed?

Reply:
It does not *need* to be added, but we could add it. As it is, it would fail with a message that not all packages were found before installing anything.

Reply:
Failing is desirable so as to avoid confusion about what installed vs failed!